### PR TITLE
issue: 848172 Fix csbuild scan report

### PR DESCRIPTION
**Pull request by liranoz12**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/396/ for details (Mellanox internal link).

File: src/state_machine/sm.cpp [hunk 1]
```diff
@@ -175,6 +176,8 @@ BULLSEYE_EXCLUDE_BLOCK_START
 					return ERROR;
 				}
 
+				SM_ASSERT_POINTER(m_p_sm_table[st].event_info);
```
Comment (rosenbaumalex) [inline src/state_machine/sm.cpp]: `SM_ASSERT_POINTER(m_p_sm_table[st].event_info)` is called immediately after the calloc() call (Line #119). 
But you think it is still needed here?

File: src/stats/stats_reader.cpp [hunk 2]
```diff
@@ -332,11 +340,12 @@ void print_basic_stats(socket_stats_t* p_stats)
 	// 
 	// Socket statistics
 	//
-	double rx_poll_hit = (double)p_stats->counters.n_rx_poll_hit;
 	double rx_poll_hit_percentage = 0;
 	
-	if (rx_poll_hit)
+	if (p_stats->counters.n_rx_poll_hit) {
```
Comment (rosenbaumalex) [inline src/stats/stats_reader.cpp]: do you also need to check `if (p_stats)` is not NULL?

File: src/stats/stats_reader.cpp [hunk 3]
```diff
@@ -356,11 +365,12 @@ void print_medium_total_stats(socket_stats_t* p_stats)
...
```
Comment (rosenbaumalex) [inline src/stats/stats_reader.cpp]: Should you also need to check if (p_stats) is not NULL?

File: src/stats/stats_reader.cpp [hunk 4]
```diff
@@ -565,15 +575,16 @@ void print_full_iomux_stats(const char* func_name, iomux_func_stats_t* p_iomux_s
...
 {
-       double iomux_poll_hit = (double)p_iomux_stats->n_iomux_poll_hit;
        double rx_poll_hit_percentage = 0;
        char post_fix[3] = "";
 
        if (user_params.print_details_mode == e_deltas)
                strcpy(post_fix, "/s");
 
-       if (iomux_poll_hit)
+       if (p_iomux_stats->n_iomux_poll_hit) {
```
Comment (rosenbaumalex) [inline src/stats/stats_reader.cpp]: don't you need to check `if (p_iomux_stats)` is not NULL?

**Comment by Di0gen** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/413/ for details (Mellanox internal link).

File: src/utils/rdtsc.cpp [hunk 5]
```diff
@@ -69,7 +69,8 @@ bool get_cpu_hz(double &hz_min, double &hz_max)
 		if (rc != 1) {
 			continue;
 		}
-		if (mhz_tmp == 0.0) {
+		// Coverity warning : comparing floating point with == or != is unsafe
+		if (0 == static_cast<int>(mhz_tmp * 100000)) {
```
Comment (rosenbaumalex) [inline src/utils/rdtsc.cpp]: converting to `int` is not the best way to do this.

1. Use a dedicate `bool do_once` flag to modify the first off min + max values

2. Use something like: `bool compareDouble(doubel a, doube b) {return fabs(a - b) < std::numeric_limits<double>::epsilon(); } `
(See referance in: http://stackoverflow.com/questions/17333/most-effective-way-for-float-and-double-comparison)

File: src/vma/event/event.h [hunk 6]
```diff
@@ -35,6 +35,7 @@
 #define EVENT_H
 
 #include "vma/util/to_str.h"
+#include "utils/bullseye.h"
```
Comment (rosenbaumalex) [inline src/vma/event/event.h]: nicer to keep include order as follows:
1. system includes on top (<typeinfo, stdio, stdint...)
2. general product includes second (utils/bullseye.h))
3. specific module includes last (vma/util/to_str.h)

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/397/ for details (Mellanox internal link).

File: src/vma/main.cpp [hunk 7]
```diff
@@ -264,7 +264,7 @@ void check_cpu_speed()
 		vlog_printf(VLOG_DEBUG, "Verify with: cat /proc/cpuinfo | grep \"MHz\\|clock\"\n");
 		vlog_printf(VLOG_DEBUG, "***************************************************************************\n");
 	}
-	else if (hz_min != hz_max) {
+	else if (hz_max > hz_min) {
```
Comment (rosenbaumalex) [inline src/vma/main.cpp]: better to use idea of `double` comparator from above

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/414/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/398/ for details (Mellanox internal link).

File: src/state_machine/sm.cpp (hunk 1 above)
Comment (liranoz12) [inline src/state_machine/sm.cpp]: The error does not disappear when i remove it.

File: src/stats/stats_reader.cpp (hunk 2 above)
Comment (liranoz12) [inline src/stats/stats_reader.cpp]: There is no error regarding this line.

File: src/stats/stats_reader.cpp (hunk 3 above)
Comment (liranoz12) [inline src/stats/stats_reader.cpp]: There is no error regarding this line.

File: src/stats/stats_reader.cpp (hunk 4 above)
Comment (liranoz12) [inline src/stats/stats_reader.cpp]: There is no error regarding this line.

File: src/vma/event/event.h (hunk 6 above)
Comment (liranoz12) [inline src/vma/event/event.h]: ok

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/415/ for details (Mellanox internal link).

File: src/utils/rdtsc.cpp (hunk 5 above)
Comment (liranoz12) [inline src/utils/rdtsc.cpp]: ok

File: src/vma/main.cpp (hunk 7 above)
Comment (liranoz12) [inline src/vma/main.cpp]: ok

**Comment by liranoz12** [comment]
@rosenbaumalex  please see the last commit

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/399/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/400/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/416/ for details (Mellanox internal link).

**Comment by rosenbaumalex** [comment]
:+1:

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/401/ for details (Mellanox internal link).

**Comment by OphirMunk** [comment]
Merged #185

**Pull request closed without merging**